Yosemite/YosemiteTests/Stores/ProductStoreTests.swift

```diff
@@ -757,6 +757,123 @@ final class ProductStoreTests: XCTestCase {
         XCTAssertEqual(viewStorage.countObjects(ofType: Storage.Product.self), 0)
     }
 
+    // MARK: - ProductAction.retrieveMostPopularProductsInCache
+
+    func test_retrieveMostPopularProductsInCache_when_there_are_several_popular_products_returns_them_sorted() {
+        let firstProductID: Int64 = 1
+        let secondProductID: Int64 = 2
+        let thirdProductID: Int64 = 3
+
+        storageManager.insertSampleProduct(readOnlyProduct: Product.fake().copy(productID: firstProductID))
+        storageManager.insertSampleProduct(readOnlyProduct: Product.fake().copy(productID: secondProductID))
+        storageManager.insertSampleProduct(readOnlyProduct: Product.fake().copy(productID: thirdProductID))
+
+        let firstOrderItem = OrderItem.fake().copy(productID: firstProductID)
+        let secondOrderItem = OrderItem.fake().copy(productID: secondProductID)
+        let thirdOrderItem = OrderItem.fake().copy(productID: thirdProductID)
+
+        let firstOrder = Order.fake().copy(siteID: sampleSiteID, status: .completed)
+        let secondOrder = Order.fake().copy(siteID: sampleSiteID, status: .completed)
+        let thirdOrder = Order.fake().copy(siteID: sampleSiteID, status: .completed)
+
+        storageManager.insertSampleOrder(readOnlyOrder: firstOrder).items = [storageManager.insertSampleOrderItem(readOnlyOrderItem: firstOrderItem)]
+        storageManager.insertSampleOrder(readOnlyOrder: secondOrder).items = [storageManager.insertSampleOrderItem(readOnlyOrderItem: firstOrderItem),
+                                                                              storageManager.insertSampleOrderItem(readOnlyOrderItem: secondOrderItem)]
+        storageManager.insertSampleOrder(readOnlyOrder: thirdOrder).items = [storageManager.insertSampleOrderItem(readOnlyOrderItem: firstOrderItem),
+                                                                             storageManager.insertSampleOrderItem(readOnlyOrderItem: secondOrderItem),
+                                                                             storageManager.insertSampleOrderItem(readOnlyOrderItem: thirdOrderItem)]
```

Comment:
Perhaps we could use some factory methods to help with this, as we're repeating them in all tests:

```
[firstProductID, secondProductID, thirdProductID].forEach { productID in
  storageManager.insertSampleProduct(readOnlyProduct: Product.fake().copy(productID: productID))
  // makeOrderItem()
}
  
// makeOrder(with: [OrderItem]) -> Order
// insertSampleOrderInStorage(with: Order)
...
```

Reply:
Good idea, done in bc1208c435c3f26ccb6db8a2aeb3ea8cf66693f8